src/server/pps/server/api_server.go

```diff
@@ -3012,12 +3012,21 @@ func (a *apiServer) listPipeline(ctx context.Context, request *pps.ListPipelineR
 			})
 		})
 	}
+	var offset int64
+	if request.Page != nil {
+		offset = request.Page.PageIndex * request.Page.PageSize
```

Comment:
You can reduce this whole `var … if … { … }`  to `offset := request.GetPage().GetPageIndex() * request.GetPage().GetPageSize()`. It _is_ mildly ugly to call `request.GetPage` twice, but it’s not the end of the world, and you could always assign it to a variable if you like, and then use that variable down at line 3,027.